resources/baseResource.ts

```diff
@@ -62,6 +62,26 @@ export class BaseResource {
             .catch(error => { throw extractUnitError(error) })
     }
 
+    protected async httpPatchFullPath<T>(path: string, data: DataPayload | { data: DataPayload; }, config?: { headers?: object; params?: object; }): Promise<T> {
+        const conf = {
+            headers: this.mergeHeaders(config?.headers),
+            maxBodyLength: MAX_REQUEST_SIZE,
+            maxContentLength: MAX_REQUEST_SIZE,
+            ...(config?.params && { params: (config.params) })
+        }
+
+        const d = !data || (data && "data" in data) ? data : {
+            data: {
+                type: data.type,
+                attributes: data.attributes
+            }
+        }
+
+        return await this.axios.patch<T>(path, d, conf)
+            .then(r => r.data)
+            .catch(error => { throw extractUnitError(error) })
+    }
+
```

Comment:
`httpPatch` and `httpPatchFullPath` are the same function except for
 `return await this.axios.patch<T>(this.resourcePath + path, d, conf)` 
 vs
 `return await this.axios.patch<T>(path, d, conf)`
 Let's have them re-use one another